README.md

```diff
@@ -89,6 +89,7 @@ optional arguments:
   -h, --help            show this help message and exit
   --sys-config SYS_CONFIG
                         Path to the pre generated sys-config json
+  --list_fw_ver         List the current version of firmware and exit
```

Comment:
Instead of an option could you add the fw-version list as a seperate command.

So you would run something like `tt-flash version` to list the versions. That way it could be extended to show more information in the future.

Reply:
@TTDRosen sadly that code already has that option. It looks to be an incomplete implementation that only shows the version of the `tt-flash` program and has nothing to do with the firmware running on the cards.

The `tt-flash` code really needs to be refactored. I looked at creating a separate command, however, it would end up just copying a bunch of code nested deep in the actual flash function.

IMO the best approach would be to have some shared, cleaned up code between `tt-flash` and `tt-smi` that had compatible command line options to get this information. `tt-smi` has a very clean way to display firmware version in [TTDataTable](https://github.com/tenstorrent/tt-smi/blob/main/tt_smi/tt_smi.py#L152).

`tt-flash` needs to be refactored and already displays firmware versions if you try to flash. It has no way to tell you the current version so I made this simple command line change to leverage what already exists with minimal changes.

Maybe @sbansalTT can provide feedback on the best approach to get things in a consistent state?

### tt-smi Device Firmware Versions
![tt-smi-fw-vers](https://github.com/user-attachments/assets/9c2f79fb-22dd-40fd-9813-1cf1bd8ce946)

Reply:
Yeah, I agree that to properly add the version display we should refactor tt-flash rather than trying to fit it into a structure that wasn't designed for an early exit. However for the purposes of flashing I think that showing all running fw is unnecessary as that is more a debugging/information detail rather than something relavent for flashing. Instead I think that it would make more sense to just display the running bundle version. Which would be accessed by a separate command, for this I don't mind if you delete the existing version command and replace it with a new handler.